app/scripts/lib/transaction/eip5792.ts

```diff
@@ -0,0 +1,66 @@
+import { SendCalls } from '@metamask/eth-json-rpc-middleware/dist/methods/wallet-send-calls';
+import { NetworkController } from '@metamask/network-controller';
+import { rpcErrors } from '@metamask/rpc-errors';
+import { TransactionController } from '@metamask/transaction-controller';
+import { Hex, JsonRpcRequest } from '@metamask/utils';
+
+export async function processSendCalls(
+  transactionController: TransactionController,
+  networkController: NetworkController,
```

Comment:
Might it make more sense / match our existing patterns better to pass in hooks to the actual methods you're calling rather than the full controllers?

Reply:
Good point, it definitely would.

In a future PR, we could even just pass in the messenger and invoke the controllers using the messenger actions.

Reply:
Done.